i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/gang_scheduling.md

```diff
@@ -0,0 +1,608 @@
+
+因此，核心中的处理需要考虑两种会影响特定状态的转换的情况：
+1. 占位请求挂起（退出 _accepted_ ）
+2. 占位分配（退出 _waiting_ ）
+
+占位请求挂起：
+挂起的占位符是通过超时来处理的。
+应用程序只能花费有限的时间等待分配所有预留资源。
+这个超时是必需的，因为应用程序的部分占位分配可能会占用集群资源而没有真正使用它们。
+
+应用程序可能排队等待未知时间，等待占位分配开始。
+因此，占位请求的超时不能与应用程序或请求的创建相关联。
+超时必须在分配第一个占位请求时开始。
+
+在所有占位请求都被分配之前，应用程序不能请求真正的分配。
+占位请求也由 shim 跟踪，因为它代表一个 pod。
+在核心中释放请求需要消息在核心和 shim 之间流动以释放该请求。
+但是，在这种情况下，分配预留资源的超时会触发应用程序失败。
+当超时被触发并且占位请求挂起时，应用程序将从它所处的状态转换为 _killed_ 。
+
+本案例的应用状态可以概括为：
+*   申请状态为 _accepted_
+*   占位分配的资源大于零，并且小于 _AddApplicationRequest_ 中的 _placeholderAsk_
+*   待处理资源请求大于零
+
+进入 _killed_ 状态必须自动将应用程序移出队列。
+
+状态更改和占位分配释放可以在单个更新响应消息中处理。 该消息将包含以下内容：
+* _UpdatedApplication_ 用于应用程序的状态变化
+* 一个或多个 _AllocationRelease_ 消息，每个占位一个，_TerminationType_ 设置为 TIMEOUT
+* 一条或多条 _TerminationType_ 设置为 TIMEOUT 的 AllocationAskRelease 消息
+
+shim 首先处理 AllocationAskRelease 消息，然后是 _AllocationResponse_ 消息，最后是 _UpdatedApplication_ 消息。应用程序状态在核心端更改为 _killed_ 状态仅取决于移除所有占位 pod，而不取决于对 _UpdatedApplication_ 消息的响应。
+
+![占位超时](./../assets/gang_timeout.png)
+
+占位超时期间 shim 和核心的组合流程：
+*   核心超时占位分配。 (1)
+*   删除占位分配被传递到 shim。 (2)
+*   所有占位分配都由 shim 释放，并传回核心。
+*   删除占位分配请求被传递给 shim。 (3)
+*   所有占位分配请求都由 shim 释放，并传回核心。
+*   在占位分配和请求被释放后，核心将应用程序移动到终止状态，将其从队列中移除 (4)。
+*   状态更改在核心和 shim 中完成。 (5) 
+
+
+已分配的占位：
+剩余的占位需要由核心释放。
+需要通知 shim 将它们移除。这必须在进入 _完成_ 状态时触发。
+在核心请求释放占位之后，应用程序的状态转换可以继续。
+核心将处理从 shim 返回的占位分配的 _AllocationRelease_ 消息，并将 _TerminationType_ 设置为正常*超时*，而不会触发状态更改。
+
+状态更改和占位分配释放可以在单个更新响应消息中处理。
+该消息将包含以下内容：
+*   _UpdatedApplication_ 用于应用程序的状态变化
+*   零个或多个 _AllocationRelease_ 消息，每个占位一个，_TerminationType_ 设置为 *超时*
+
+shim 首先处理 _AllocationResponse_ 消息，然后是 _UpdatedApplication_ 消息。
+应用程序状态在核心端更改为 _完成_ 状态仅依赖于所有占位 pod 的移除，而不依赖于对 _UpdatedApplication_ 消息的响应。
+
+进入 _完成_ 状态会自动将应用程序移出队列。
+这也应该处理我们之前讨论的可能延迟处理来自 shim 的请求的情况，因为我们可以在需要时从 _等待_ 移回 _执行中_ 。
+一个 _完成_ 的应用程序也不应该阻止围绕cron讨论的情况，例如每次调用使用相同的应用程序ID提交。
+具有相同申请ID的 _完成_ 申请不得阻止提交具有相同ID的新申请。 
+
+![应用程序清理流程](./../assets/gang_clean_up.png)
+cation to the completed state removing it from the queue (6).
+*   The state change is finalised in the core and shim. (7)
+在清理应用程序期间 shim 和核心的组合流程：
+*   在 Kubernetes 层发布了一个 pod。 (1)
+*   shim 将分配释放传递给核心。 (2)
+*   如果没有挂起或活动的分配，核心会将应用程序转换为等待状态。 (3)
+*   等待状态超时并触发清理。 (4)
+*   删除占位符配被传递到 shim。 (5)
+*   所有占位分配都由 shim 释放，并传回核心。
+*   释放所有占位后，核心将应用程序移动到完成状态，将其从队列中删除 (6)。
+*   状态更改在核心和 shim 中完成。 (7) 
+
+## 应用恢复
+在应用程序恢复期间，占位 Pod 与节点上的任何其他 Pod 一样被恢复。
+这些 pod 由 shim 作为节点现有分配的一部分传递到核心。
+现有的分配在核心中没有相应的 _分配请求_。 核心根据恢复的信息生成 _分配请求_ 。
+
+对于联邦调度，_分配请求_ 包含 _任务组名称_ 和 _占位_ 标志。
+在恢复期间，相同的信息必须是 _分配_ 消息的一部分。
+这是因为在两个方向上使用相同的消息，从 RM 到调度器，反之亦然，这意味着我们需要更新消息及其处理。
+
+如果 _分配_ 消息中缺少信息，则恢复的分配将不会在核心中被正确标记。
+恢复的分配将被视为常规分配。
+这意味着它作为替换占位的正常分配周期的一部分被跳过。
+
+逻辑变化只需要现有分配的恢复将接口消息中的字段复制到内核中的分配对象中。
+
+## 接口变化
+需要对 shim 和核心之间的通信进行多次更改，以支持所需的帮派信息。
+
+应用程序必须提供占位请求的总大小，以防止接受永远无法运行的应用程序。
+
+从 shim 发送到核心以进行分配请求的当前对象在 AllocationAsk 中定义。
+由于从调度器核心传回的结果消息不会更改，因此分配不会更改。。但是，对于使用相同分配消息从 shim 到核心的恢复，必须包含与帮派相关的字段。
+必须将与帮派相关的字段添加到这两个消息中。
+
+分配释放请求和响应请求需要支持双向通信，并且需要进行重大更改。
+
```

Comment:
I assume the English content below is yet to be translated?

Reply:
@yuchaoran2011 
是的，这里有两个原因。
第一个是我在翻译 gang scheduling 的时候遇到了一些困难。在我翻译的过程中发现这篇文章里面存在着很多我难以理解的句子，我结合上下文也很难翻译和猜测出来原作者的意思；同时，这里面有一些单词拼接的专有词汇我还是有一些拿捏不准，不知道是完全翻译成中文还是保留原样，因为可能这些单词在实际的代码或者使用过程中可能会有一些语义在里面，翻译成中文后反倒会让读了文章的使用者对应不上关系。
第二个是我原来的翻译顺序是按照我自己文档目录树的顺序翻译的，但是考虑到读者的阅读顺序，从website开始按照官网的顺序进行翻译可能更加好。所以我将当时翻译一半的内容先停了下来。
我个人的想法是在后续的会议中如果我参与，可能会请教一下各位前辈一些专有名词的翻译。其实有很多的专有名词已经约定俗成，在很多情况下不翻译可能让读者更好地进行理解。

当前的翻译并不是全部的，为了保证文档的完整性，我暂时将未完成的部分也追加到目录下，以防止发生文档找不到的情况

Reply:
Thanks for the clarification. I agree it's hard to find counterparts in Chinese for some terminologies. You are welcome to join the [weekly community meeting](https://yunikorn.apache.org/community/get_involved) for Mandarin speakers to discuss with people. It's at 1pm China Standard Time on Wednesday.

Reply:
hi @cdmikechen thank you so much for initiating the effort!! and thanks for @yuchaoran2011 's help to review.
I do not think we need to wait for everything ready before committing patches, we can change YUNIKORN-1029 to an umbrella and do this incrementally. Actually, I am gonna propose in the next community, we should do a split for the reviewers for people who can help to review. 

for your questions
> 第一个是我在翻译 gang scheduling 的时候遇到了一些困难
Understood, that's why we need to work together more closely : )

> 第二个是我原来的翻译顺序是按照我自己文档目录树的顺序翻译的，但是考虑到读者的阅读顺序，从website开始按照官网的顺序进行翻译可能更加好。所以我将当时翻译一半的内容先停了下来。
I think we should focus on the user-facing doc first. such as quick start, download, roadmap, community, user guide etc. for developer-facing docs, that can come later, such as design docs.

Reply:
Thanks @cdmikechen for this great contribution!

+1 if we could split this PR into smaller ones. It will be much easier for more reviewers to help on the review process. :)